File: Frontend/src/app/models/invoice.ts
import { InvoiceDetails } from './details';

export class Invoice {
  invoiceId: number = 0;
  locationId: number = 0;
  invoiceNumber: string = '';
  invoiceDate: Date = new Date();
  customerName: string = '';
  invoiceDetails: Partial<InvoiceDetails> = new InvoiceDetails();
  constructor(init?: Partial<Invoice>) {
    Object.assign(this, init);
  }
}

File: Backend/InvoiceAPI/BusinessLogic/Models/Invoice.cs
using Microsoft.EntityFrameworkCore;

namespace InvoiceAPI.BusinessLogic.Models
{
    [PrimaryKey(nameof(InvoiceId), nameof(LocationId))]
    public class Invoice
    {
        public int InvoiceId { get; set; }
        public int LocationId { get; set; }
        public string InvoiceNumber { get; set; }
        public DateTime InvoiceDate { get; set; }
        public string CustomerName { get; set; }
        public virtual ICollection<InvoiceDetails>? InvoiceDetails { get; set; }
    }
}
